### PR TITLE
[US749252] MAS property for enableSSLPinning, isSSLPinningEnabled and…

### PR DESCRIPTION
**Pull request by ysd24**
[US749252] - Flags to disable ssl pinning.

### CODE REVIEW

File: MASFoundation/Classes/MAS.h [hunk 1]
```diff
@@ -159,6 +159,26 @@
 
 
 
+/**
+*  Sets BOOL indicator of  enforcing SSL Pinning i.e. Evaluate Server Trust with pinned Server Certificates.
+*  By default, SSL Pinning is disabled.
+*
+*  @param enable BOOL value of indicating whether SSL Pinning should be enabled or not.
+*/
++ (void)enableSSLPinning:(BOOL)enable;
```
Comment (graju256) [inline MASFoundation/Classes/MAS.h]: [**optional**] Assuming that the internal property name is sslPinningEnabled,
the proper method could be: setSSLPinningEnabled(boolean).

But if you choose enableSSLPinning, avoid accepting the argument. Of course, to disable, we might require a separate method, i.e., disableSSLPinning.

File: MASFoundation/Classes/MAS.h [hunk 2]
```diff
@@ -159,6 +159,26 @@
...
+
+
+
+/**
+*  Gets BOOL indicator of SSL Pinning enabled or not for Server Trust evluation of pinned Server Certificates.
+*  By default, SSL Pinning is disabled.
+*
+*  @return BOOL value of indicating whether SSL Pinning is enabled or not.
+*/
++ (BOOL)isSSLPinningEnabled;
```
Comment (graju256) [inline MASFoundation/Classes/MAS.h:178]: This is good.

File: MASFoundation/Classes/MAS.m [hunk 3]
```diff
@@ -623,6 +635,17 @@ + (void)startWithURL:(NSURL *)url completion:(MASCompletionErrorBlock)completion
             
             NSURLSessionAuthChallengeDisposition disposition = NSURLSessionAuthChallengePerformDefaultHandling;
             
+            //
+            // Bypass SSL Pinning if its disabled.
+            //
+            if (![MAS isSSLPinningEnabled]) {
```
Comment (graju256) [inline MASFoundation/Classes/MAS.m]: Should we reset the security config over MASConfiguration?
`
 [MASConfiguration setSecurityConfiguration:securityConfig error:nil];`

Comment (ysd24) [reply MASFoundation/Classes/MAS.m]: Here, we are in the SDK start method, by this time we have not kicked in the SDK services.
And the MASSecurityPolicy is extracting the MASSecurityConfiguration per host basis. So we need to overwrite here.

File: MASFoundation/Classes/models/MASConfiguration.m [hunk 4]
```diff
@@ -889,6 +891,22 @@ - (BOOL)enabledTrustedPublicPKI
     return [mobileInfo[MASMobileEnbaledTrustedPublicPKI] boolValue];
 }
 
+
+- (BOOL)sslpinningEnabled
```
Comment (graju256) [inline MASFoundation/Classes/models/MASConfiguration.m]: can you please double-check the naming convention here?
The right one could be: ssl**P**inningEnabled

File: MASFoundation/Classes/MAS.h (hunk 1 above)
Comment (ysd24) [reply MASFoundation/Classes/MAS.h]: Done.

**Review by zakirjt — DISMISSED** [review]

**Review by arindamhit — DISMISSED** [review]

**Review by arindamhit — APPROVED** [review]

**Pull request merged**